comps/llms/text-generation/tgi/llm.py

```diff
@@ -23,16 +23,30 @@
     statistics_dict,
 )
 from comps.cores.proto.api_protocol import ChatCompletionRequest
+import json
+from typing import Optional
 
 logger = CustomLogger("llm_tgi")
 logflag = os.getenv("LOGFLAG", False)
 
-llm_endpoint = os.getenv("TGI_LLM_ENDPOINT", "http://localhost:8080")
-llm = AsyncInferenceClient(
-    model=llm_endpoint,
-    timeout=600,
-)
+# Environment variables
+MODEL_CONFIGS = os.getenv("MODEL_CONFIGS", "{}")
+DEFAULT_ENDPOINT = os.getenv("TGI_LLM_ENDPOINT", "http://localhost:8080")
 
+def get_model_endpoint(model_name: Optional[str], default_endpoint: str, model_configs: str) -> str:
```

Comment:
1. Let move this func to init flow so that we do not need to parse json on each generate call. 
2. Also support providing the json as a file read.